library/core/src/ops/control_flow.rs

```diff
@@ -56,6 +56,20 @@ impl<B, C> ControlFlow<B, C> {
             ControlFlow::Break(x) => Some(x),
         }
     }
+
+    /// Maps `ControlFlow<B, C>` to `ControlFlow<T, C>` by applying a function
+    /// to the break value in case it exists.
+    #[inline]
+    #[unstable(feature = "control_flow_enum", reason = "new API", issue = "75744")]
+    pub fn map_break<T, F>(self, f: F) -> ControlFlow<T, C>
+    where
+        F: FnOnce(B) -> T,
+    {
+        match self {
+            ControlFlow::Continue(x) => ControlFlow::Continue(x),
+            ControlFlow::Break(x) => ControlFlow::Break(f(x)),
+        }
+    }
```

Comment:
I wonder which combinators `ControlFlow` should have, there could be different flavours of `and`s, `or`s, `map`s, `flatten`, `as_ref`...

Reply:
don't really know. `map_break` and `map_continue` probably make sense.

The others depending on if they are needed.